### ISSUE TITLE
SIGSEGV when importing scala.collection.Iterator$Leading$1

### ISSUE THREAD

**Issue by KOLANICH** [jpype-project/jpype#841]
JPype version: 23f3c95180c35ed3c7cb2f6a59bac4aa5b493b0d

`scala/focal,focal,now 2.11.12-4 all [installed]`

```python
import jpype
from pathlib import Path
scalaPath = Path("/usr/share/scala-2.11/lib")
jpype.startJVM(classpath=[str(scalaPath / "scala-library.jar")])
jpype.JClass("scala.collection.Iterator$Leading$1")
```

```
A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f21fe1f9b94, pid=1316593, tid=1316593
#
# JRE version: OpenJDK Runtime Environment (14.0.1+7) (build 14.0.1+7-Ubuntu-1ubuntu1)
# Java VM: OpenJDK 64-Bit Server VM (14.0.1+7-Ubuntu-1ubuntu1, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x25cb94]

Current thread (0x0000000002f91000):  JavaThread "main" [_thread_in_vm, id=1316593, stack(0x00007ffc842d3000,0x00007ffc843d3000)]

Stack: [0x00007ffc842d3000,0x00007ffc843d3000],  sp=0x00007ffc843cb488,  free space=993k
Native frames: (J=compiled Java code, A=aot compiled Java code, j=interpreted, Vv=VM code, C=native code)
V  [libjvm.so+0x25cb94]
C  [_jpype.cpython-38-x86_64-linux-gnu.so+0x5dcbb]  JPJavaFrame::GetStringUTFChars(_jstring*, unsigned char*)+0x1b

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  org.jpype.manager.TypeFactoryNative.defineObjectClass(JLjava/lang/Class;Ljava/lang/String;J[JI)J+0
j  org.jpype.manager.TypeManager.createOrdinaryClass(Ljava/lang/Class;ZZ)Lorg/jpype/manager/ClassDescriptor;+304
J 368 c1 org.jpype.manager.TypeManager.getClass(Ljava/lang/Class;)Lorg/jpype/manager/ClassDescriptor; (31 bytes) @ 0x00007f21e02a8f4c [0x00007f21e02a8b80+0x00000000000003cc]
j  org.jpype.manager.TypeManager.findClass(Ljava/lang/Class;)J+106
j  org.jpype.manager.TypeManager.findClassByName(Ljava/lang/String;)J+14
v  ~StubRoutines::call_stub
```

<details>
<summary> Also these classes cause crashes:</summary>

```python
sc = "scala."
scc = sc + "collection."
scrf = sc + "reflect."
scrfi = scrf + "internal."
scrfo = scrf + "io."
sct = sc + "tools."
sctn = sct + "nsc."
sctnbj = sctn + "backend.jvm."
sctnc = sctn + "typechecker."
sctnt = sctn + "transform."
sctntp = sctnt + "patmat."
sctni = sctn + "interpreter."
scrfit = scrfi + "tpe."
scrfm = scrf + "macros."
scrfmc = scrfm + "contexts."

blackList = {
	scc + "Iterator$Leading$1",
	scc + "Iterator$PartitionIterator$1",
	scc + "Iterator$Partner$1",
	scc + "parallel.mutable.ParHashMapCombiner$table$2$",
	scrfi + "ExistentialsAndSkolems$$anonfun$packSymbols$1",
	scrfi + "ExistentialsAndSkolems$Deskolemizer$1",
	scrfi + "Internals$SymbolTableInternal$changeOwnerAndModuleClassTraverser$2$",
	scrfit + "TypeMaps$InstantiateDependentMap$StabilizedArgTp$",
	scrfit + "TypeMaps$InstantiateDependentMap$StableArgTp$",
	scrfit + "TypeMaps$InstantiateDependentMap$UnstableArgTp$",
	scrfit + "TypeMaps$InstantiateDependentMap$treeTrans$2$",
	scrfit + "TypeMaps$SubstSymMap$mapTreeSymbols$$anonfun$transform$2",
	scrfit + "TypeMaps$SubstSymMap$mapTreeSymbols$$anonfun$transform$1",
	scrfit + "TypeMaps$SubstTypeMap",
	scrfit + "TypeMaps$SubstTypeMap$trans$2$",
	scc + "immutable.ChampBaseIterator",
	scrfo + "FileZipArchive$FileEntry$1",
	scrfo + "JavaToolsPlatformArchive$$anonfun$iterator$2$FileEntry$4",
	scrfo + "ManifestResources$$anonfun$iterator$1$FileEntry$3",
	scrfo + "URLZipArchive$EmptyFileEntry$1",
	scrfo + "URLZipArchive$FileEntry$2",
	scrfm + "compiler.Validators$Validator$SigGenerator$2$",
	scrfm + "compiler.Validators$Validator$SigGenerator$2$SigmaTypeMap$",
	scrfmc + "Internals$$anon$1$HofTransformer",
	scrfmc + "Internals$$anon$1$HofTypingTransformer",
	scrfmc + "Reifiers$utils$2$",
	sctn + "Global$ClassPathOrdering$2$",
	sctnbj + "GenASM$JPlainBuilder$$anonfun$isClosureApply$1$1",
	sctnbj + "GenASM$JPlainBuilder$Interval$3",
	sctnbj + "GenASM$JPlainBuilder$Interval$4$",
	sctnbj + "GenASM$JPlainBuilder$LineNumberEntry$3",
	sctnbj + "GenASM$JPlainBuilder$LineNumberEntry$4$",
	sctnbj + "GenASM$JPlainBuilder$LocVarEntry$3",
	sctnbj + "GenASM$JPlainBuilder$LocVarEntry$4$",
	sctnbj + "GenASM$JPlainBuilder$scoping$2$",
	sctnbj + "opt.CallGraph$CallsiteInfo$3",
	sctnbj + "opt.CallGraph$CallsiteInfo$4$",
	sctnbj + "opt.ClosureOptimizer$closureInitOrdering$2$",
	sctn + "doc.model.IndexModelFactory$$anon$1$result$2$",
	sctn + "doc.model.ModelFactory$EntityImpl$$anonfun$annotations$1$$typecreator1$1",
	sctn + "interactive.REPL$compiler$2$",
	sctni + "IMain$$typecreator2$1",
	sctni + "Imports$ReqAndHandler$3",
	sctni + "Imports$ReqAndHandler$4$",
	sctni + "ReplVals$AppliedTypeFromTags$1",
	sctn + "symtab.BrowsingLoaders$BrowserTraverser$1",
	sctnt + "Constructors$OmittablesHelper$detectUsages$2$",
	sctnt + "Mixin$MixinTransformer$AddInitBitsTransformer$1",
	sctnt + "Mixin$MixinTransformer$AddInitBitsTransformer$1$$anonfun$transformStats$1",
	sctnt + "Mixin$MixinTransformer$TreeSymSubstituterWithCopying$1",
	sctnt + "Mixin$SingleUseTraverser$2$",
	sctnt + "SpecializeTypes$$anon$2$CollectMethodBodies",
	sctnt + "SpecializeTypes$FullTypeMap$1",
	sctntp + "Interface$TypedSubstitution$Substitution$substIdentsForTrees$2$",
	sctntp + "Logic$PropositionalLogic$gatherEqualities$2$",
	sctntp + "Logic$PropositionalLogic$rewriteEqualsToProp$2$",
	sctntp + "MatchAnalysis$MatchAnalyzer$VariableAssignment$3",
	sctntp + "MatchAnalysis$MatchAnalyzer$VariableAssignment$3$$anonfun$uniqueEqualTo$1$$anonfun$apply$20",
	sctntp + "MatchAnalysis$MatchAnalyzer$VariableAssignment$4$",
	sctntp + "MatchApproximation$MatchApproximator$TreeMakersToProps$TreeMakerToProp$condStrategy$2$",
	sctntp + "MatchTranslation$MatchTranslator$ExtractorCallRegular",
	sctntp + "MatchTranslation$MatchTranslator$ExtractorCallRegular$splice$2$",
	sctntp + "ScalaLogic$TreesAndTypesDomain$Var$ExcludedPair$2",
	sctntp + "ScalaLogic$TreesAndTypesDomain$Var$ExcludedPair$3$",
	sctntp + "Solving$Solver$TseitinSolution$3",
	sctntp + "Solving$Solver$TseitinSolution$4$",
	sctntp + "TreeAndTypeAnalysis$CheckableTreeAndTypeAnalysis$typeArgsToWildcardsExceptArray$2$",
	sctnc + "Implicits$ImplicitSearch$ImplicitComputation$LocalShadower$1",
	sctnc + "Implicits$ImplicitSearch$ImplicitComputation$LocalShadower$1",
	sctnc + "Implicits$ImplicitSearch$ImplicitComputation$NoShadower$2$",
	sctnc + "Infer$Inferencer$InferMethodAlternativeTwice$1",
	sctnc + "Infer$Inferencer$InferTwice$1",
	sctnc + "Macros$UnsigmaTypeMap$2$",
	sctnc + "RefChecks$RefCheckTransformer$LevelInfo",
	sctnc + "RefChecks$RefCheckTransformer$MixinOverrideError$3",
	sctnc + "RefChecks$RefCheckTransformer$MixinOverrideError$4$",
	sctnc + "Typers$Typer$ArrayInstantiation$2$",
	sctnc + "Typers$Typer$checkEphemeralDeep$2$",
	sct + "reflect.FormatInterpolator$$typecreator1$1",
}
```

</details>

**Comment by Thrameos** [comment]
My development match is unfortunately still down.   However I can verify that I was able to replicate the problem replicates for me.

```
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007fffff5d5859 in __GI_abort () at abort.c:79
#2  0x00007ffffdded5a5 in ?? () from /usr/lib/jvm/java-11-openjdk-amd64/lib/server/libjvm.so
#3  0x00007ffffe018257 in ?? () from /usr/lib/jvm/java-11-openjdk-amd64/lib/server/libjvm.so
#4  0x00007ffffe018b1b in ?? () from /usr/lib/jvm/java-11-openjdk-amd64/lib/server/libjvm.so
#5  0x00007ffffe018b4e in ?? () from /usr/lib/jvm/java-11-openjdk-amd64/lib/server/libjvm.so
#6  0x00007ffffddf85c0 in JVM_handle_linux_signal () from /usr/lib/jvm/java-11-openjdk-amd64/lib/server/libjvm.so
#7  0x00007ffffddebc68 in ?? () from /usr/lib/jvm/java-11-openjdk-amd64/lib/server/libjvm.so
#8  <signal handler called>
#9  0x00007ffffd75deb1 in ?? () from /usr/lib/jvm/java-11-openjdk-amd64/lib/server/libjvm.so
#10 0x00007ffffdae7cd9 in ?? () from /usr/lib/jvm/java-11-openjdk-amd64/lib/server/libjvm.so
#11 0x00007ffffe9e6966 in JNIEnv_::GetStringUTFChars (isCopy=0x7ffffffece68 "@\317\376\377\377\177", str=0x7ffffffecf40, this=<optimized out>)
    at /usr/lib/jvm/java-11-openjdk-amd64/include/jni.h:1622
#12 JPJavaFrame::GetStringUTFChars (this=this@entry=0x7ffffffecf40, a0=a0@entry=0x0, a1=a1@entry=0x7ffffffeced8 "")
    at native/common/jp_javaframe.cpp:940
#13 0x00007ffffe9e6c7f in JPStringAccessor::JPStringAccessor (jstr=<optimized out>, frame=..., this=0x7ffffffeced0)
    at native/common/jp_javaframe.cpp:1051
#14 JPJavaFrame::toStringUTF8[abi:cxx11](_jstring*) (this=this@entry=0x7ffffffecf40, str=str@entry=0x0) at native/common/jp_javaframe.cpp:1077
#15 0x00007ffffe9f6288 in JPTypeFactory_defineObjectClass (env=<optimized out>, self=<optimized out>, contextPtr=10237648, cls=0x7ffffffed288,
    name=0x0, superClass=15203680, interfacePtrs=0x7ffffffed268, modifiers=1) at native/common/jp_typefactory.cpp:173
#16 0x00007fffe525f990 in ?? ()
#17 0x00007ffffffed268 in ?? ()
#18 0x0000000400000001 in ?? ()
#19 0x00007ffffffed1c0 in ?? ()
#20 0x00000008009a0034 in ?? ()
#21 0x00007ffffffed228 in ?? ()
#22 0x000000080035a6b0 in ?? ()
#23 0x0000000000000000 in ?? ()
```

The issue is that class you are attempting to create has no name.   It is a two line fix to force it to use a different name field.  Addressed in #843

**Comment by marscher** [comment]
fixed by #843

**Comment by KOLANICH** [comment]
I confirm that #843 has fixed this issue. Thank you, @Thrameos.
